pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -100,11 +99,7 @@ func (h *handler) getArgsEnvAndStatus(infraObj *infraObject, args map[string]int
 			Name:      name2.SafeConcatName(infraObj.meta.GetName(), "machine", "driver", "secret"),
 			Namespace: infraObj.meta.GetNamespace(),
 		},
-		Data: map[string][]byte{
-			"HTTP_PROXY":  []byte(os.Getenv("HTTP_PROXY")),
-			"HTTPS_PROXY": []byte(os.Getenv("HTTPS_PROXY")),
-			"NO_PROXY":    []byte(os.Getenv("NO_PROXY")),
-		},
```

Comment:
Maybe we should always insert these in `WhitelistEnvironmentVars` to keep the current behavior. `WhitelistEnvironmentVars` will default to `NO_PROXY,HTTP_PROXY,HTTPS_PROXY` but if you set it manually to like `MACHINE_DEBUG`, you need to remember to put the proxy vars.

Reply:
I think it's fine to keep the current solution as more flexible (user has complete control over which env vars are carried over) and matching the behavior of jailer that is used for RKE1 provisioning.